exercises/README.md

```diff
@@ -16,4 +16,4 @@ xmake -P exercises
 
 ## Exercises
 
-Currently, there is one exercise: [Hardware Access Control](./hardware_access_control/README.md)
+Currently, there is one exercise: [Hardware Access Control](./hardware_access_control/)
```

Comment:
We want to point at the exercises readme and not it's directory. I think there's an important semantic difference.

Reply:
The reason I made this change is that link currently exits the mdbook and goes to GitHub. I would like it to stay within the mdbook. Happy to fix it in another way though.

Reply:
I think the reason you actually made this change was that this used to lead to `.../README.html` that didn't exist until the mdbook-regex PR. That PR the mistakenly redirected it to github. This will be fixed by https://github.com/lowRISC/sonata-software/pull/36

Reply:
Thanks for the fix!